CCDB/include/CCDB/BasicCCDBManager.h

```diff
@@ -329,6 +333,19 @@ T* CCDBManagerInstance::getForRun(std::string const& path, int runNumber, bool s
   return getForTimeStamp<T>(path, start / 2 + stop / 2);
 }
 
+template <typename T>
+T* CCDBManagerInstance::getSpecificForRun(std::string const& path, int runNumber, MD metaData)
```

Comment:
as far as I understand, this is almost the same as `getForRun` apart from the additional metaData filter, right?

I would suggest to try to avoid the code duplication with `getForRun` by implementing `getForRun` in terms of `getSpecificForRun`.

Reply:
Hi @sawenzel, yes, fair point - I have now made `getForRun` invoke `getSpecificForRun` and avoided to duplicate the few lines for getting the mid-of-run timestamp. Let me know in case I missed something :-)

Reply:
Thanks!!